practice_app/pages/api/getProductCatalog.js

```diff
@@ -0,0 +1,278 @@
+const getProductCatalog = (req, res) =>
```

Comment:
This function is too long. Please break it down to different functions based on request type.